File: Hash Map & Set/Problem1 - 2215.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Main {
    /* brute force approach - two loops */
    /* optimal approach - using hashing/set */
    static List<List<Integer>> findDifference(List<Integer> nums1, List<Integer> nums2) {
        Set<Integer> set1 = new HashSet<>(nums1);
        Set<Integer> set2 = new HashSet<>(nums2);

        List<List<Integer>> ans = new ArrayList<>();
        ans.add(new ArrayList<>());
        ans.add(new ArrayList<>());

        for (int num : set1)
            if (!set2.contains(num))
                ans.get(0).add(num);

        for (int num : set2)
            if (!set1.contains(num))
                ans.get(1).add(num);

        return ans;
        /* TC-O(2*(n + m)) SC-O(n + m) */
    }

    public static void main(String[] args) {
        List<Integer> nums1 = List.of(1, 2, 3, 4, 5);
        List<Integer> nums2 = List.of(3, 4, 5, 6, 7);

        List<List<Integer>> result = findDifference(nums1, nums2);
        System.out.println("Difference between nums1 and nums2:");
        System.out.println("Elements in nums1 but not in nums2: " + result.get(0));
        System.out.println("Elements in nums2 but not in nums1: " + result.get(1));
    }
}

File: Hash Map & Set/Problem1 - 2215.cpp
#include <bits/stdc++.h>
using namespace std;

/* brute force approach - two loops */
/* optimal approach - using hashing/set */
vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
    unordered_set<int> set1(nums1.begin(), nums1.end()); 
    unordered_set<int> set2(nums2.begin(), nums2.end()); 

    vector<vector<int>> ans(2);  
    for(auto &it : set1) 
        if(set2.find(it) == set2.end()) ans[0].push_back(it); 
    
    for(auto &it : set2) 
        if(set1.find(it) == set1.end()) 
            ans[1].push_back(it); 
        
    return ans; 
    /* TC-O(2*(n + m)) SC-O(n + m) */
}